src/Symfony/Component/Security/Http/Logout/LogoutUrlGenerator.php

```diff
@@ -151,6 +151,10 @@ private function getListener(?string $key): array
             }
         }
 
+        if (null === $this->currentFirewallName) {
+            throw new \InvalidArgumentException('Unable to find the current firewall LogoutListener, please make sure request is behind a firewall.');
+        }
+
         throw new \InvalidArgumentException('Unable to find the current firewall LogoutListener, please provide the provider key manually.');
```

Comment:
"Unable to find logout in the current firewall, please pass the firewall name manually to generate a logout URL."